scripts/utils/setup-roles.ts

```diff
@@ -99,34 +102,47 @@ export async function main(
     // giving to user to call PromissoryNote functions directly
     for (const note of [borrowerNote, lenderNote]) {
         await note.connect(deployer).initialize(loanCore.address, {
-            gasLimit,
+            gasLimit
         });
     }
 
-    // change loancore's owner from deployer to admin
-    const updateOwner = await loanCore.grantRole(DEFAULT_ADMIN_ROLE, admin.address, {
-        gasLimit,
+    // grant LoanCore the admin role to enable authorizeUpgrade onlyRole(DEFAULT_ADMIN_ROLE)
+    const updateLoanCoreAdmin = await loanCore.grantRole(ADMIN_ROLE, ADMIN_ADDRESS, {
+        gasLimit
     });
+    await updateLoanCoreAdmin.wait();
 
-    await updateOwner.wait();
     console.log(`loanCore has granted admin role: ${ADMIN_ROLE} to address: ${ADMIN_ADDRESS}`);
 
     // grant LoanCore admin fee claimer permissions
-    const updateLoanCoreFeeClaimer = await loanCore.grantRole(FEE_CLAIMER_ROLE, admin.address, {
-        gasLimit,
+    const updateLoanCoreFeeClaimer = await loanCore.grantRole(FEE_CLAIMER_ROLE, ADMIN_ADDRESS, {
+        gasLimit
     });
-
     await updateLoanCoreFeeClaimer.wait();
+
     console.log(`loanCore has granted fee claimer role: ${FEE_CLAIMER_ROLE} to address: ${ADMIN_ADDRESS}`);
 
-    const gasPrice = ethers.utils.parseUnits("100", "gwei");
+    // grant VaultFactory the admin role to enable authorizeUpgrade onlyRole(DEFAULT_ADMIN_ROLE)
+    const updateVaultFactoryAdmin = await factory.grantRole(ADMIN_ROLE, ADMIN_ADDRESS, {
+        gasLimit
+    });
+    await updateVaultFactoryAdmin.wait();
+
+    console.log(`vaultFactory has granted admin role: ${ADMIN_ROLE} to address: ${ADMIN_ADDRESS}`);
+
+    // grant originationContoller the owner role to enable authorizeUpgrade onlyOwner
+    const updateOriginationControllerAdmin = await loanCore.connect(deployer).grantRole(ADMIN_ROLE, ADMIN_ADDRESS, {
+        gasLimit
+    });
+    await updateOriginationControllerAdmin.wait();
+
+    console.log(`originationController has granted admin role: ${ADMIN_ROLE} to address: ${ADMIN_ADDRESS}`);
 
     // grant originationContoller the originator role
     const updateOriginationControllerRole = await loanCore
-        .connect(admin)
+        .connect(deployer)
         .grantRole(ORIGINATOR_ROLE, ORIGINATION_CONTROLLER_ADDRESS, {
-            gasPrice,
-            gasLimit,
+            gasLimit
```

Comment:
If we are including a defined gas limit something is wrong - we should remove this line, and if something breaks, figure out the real problem.

Reply:
When `gasLimit` is removed, `cannot estimate gas` error is generated:

![Screen Shot 2022-06-15 at 3 16 44 PM](https://user-images.githubusercontent.com/42501101/173906946-f15ba72a-2e32-4e75-a1da-9c8ec3405bd4.png)